### ISSUE TITLE
Invorrect logging protobuf requests/responces

### ISSUE THREAD

**Issue by juncevich** [zalando/logbook#1330]
Useless converting raw bytes from request(response) bodies broke protobuf model deserialization.

## Description
Protobuf response body incorrect deserializing, because FilteredHttpRequest(FilteredHttpResponse).getBody() converts body to string and after that convert string to byte array. Which cause to brake model from body. For example, if i have raw body bytes array size 170, response.getBody() array of bytes size will 250, because it was converted to string.

## Expected Behavior
FilteredHttpRequest.getBody() gives raw bytes, without converting to string.

## Actual Behavior
<img width="438" alt="изображение" src="https://user-images.githubusercontent.com/6944077/172378970-731ef7d4-df30-460b-b6ba-1b7fd07ddefe.png">

## Possible Fix
getBody must contains raw bytes from response

**Comment by skjolber** [comment]
@juncevich out of interest, which web server are you running?

**Comment by juncevich** [comment]
@skjolber Netty with Micronaut.

**Comment by skjolber** [comment]
I wonder if gRPC logging might be better served with a gRPC-centric module. At least thats how we're doing it internally (emulating the Logbook output format).

Also the HTTP 2 does not follow a strict request-response any more, with 1-n, n-1 or n-n combinations, not sure whether logbook will work as-is now @whiskeysierra?

**Comment by juncevich** [comment]
I've offered not to convert body to string in method `byte[] getBody()`. I suppose when when someone asked body in bytes - bytes must be raw. There is no need to convert body to string and after it convert to bytes. If i want body as string a can use `String getBody()`.

**Comment by whiskeysierra** [comment]
> ## Actual Behavior
> 
> <img alt="изображение" width="438" src="https://user-images.githubusercontent.com/6944077/172378970-731ef7d4-df30-460b-b6ba-1b7fd07ddefe.png">

Is that the default implementation right now?

**Comment by juncevich** [comment]
Yep 
https://github.com/zalando/logbook/blob/c54d0cb046651201499cebb952016222d89fd3d0/logbook-core/src/main/java/org/zalando/logbook/FilteredHttpRequest.java#L78

**Comment by juncevich** [comment]
@whiskeysierra Will we fix these problems? I can do it)

**Comment by whiskeysierra** [comment]
> Will we fix these problems? I can do it)

I'd love to, but it sounds like a breaking change, tbh.

**Comment by whiskeysierra** [comment]
> Useless converting raw bytes from request(response) bodies broke protobuf model deserialization.

@juncevich Just curious, do you just use protobuf over HTTP? I'm thinking how to integrate Logbook with gRPC and I'm looking for ways to do that cleanly.

**Comment by skjolber** [comment]
FYI we are using `grpc-java` and `grpc-netty` with subclasses of `io.grpc.ServerInterceptor`.

**Comment by whiskeysierra** [comment]
Can you share some of that code by any chance?

**Comment by skjolber** [comment]
Yes. I've extracted the relevant subset of our internal logging project and published it [here](https://github.com/entur/grpc-logging), it still needs some cleanups, but it is a fully working example (at least the unit tests run). 

You'll notice that Logbook is not in use at all, however the output (JSON) format is essentially the same as in Logbook.

Also the gRPC stack typically lets us return a ton of responses for a single request, this turned out to be too noisy for our taste, so we added some "summary" request-response message types.

**Comment by juncevich** [comment]
> > Useless converting raw bytes from request(response) bodies broke protobuf model deserialization.
> 
> @juncevich Just curious, do you just use protobuf over HTTP? I'm thinking how to integrate Logbook with gRPC and I'm looking for ways to do that cleanly.

Yes.
